src/components/fields/text-field/text-field.spec.js

```diff
@@ -1,172 +1,189 @@
-    });
-    it('should disable the TextInput', () => {
-      expect(wrapper.find(TextInput)).toHaveProp('isDisabled', true);
-    });
+describe('when disabled', () => {
+  it('should disable the input', () => {
+    const { getByLabelText } = renderTextField({ isDisabled: true });
+    expect(getByLabelText('TextField')).toHaveAttribute('disabled');
   });
+});
 
-  describe('when read-only', () => {
-    let props;
-    let wrapper;
-    beforeEach(() => {
-      props = createTestProps({ isReadOnly: true });
-      wrapper = shallow(<TextField {...props} />);
-    });
-    it('should mark the TextInput as read-only', () => {
-      expect(wrapper.find(TextInput)).toHaveProp('isReadOnly', true);
-    });
+describe('when readOnly', () => {
+  it('should disable the input', () => {
+    const { getByLabelText } = renderTextField({ isReadOnly: true });
+    expect(getByLabelText('TextField')).toHaveAttribute('readonly');
   });
+});
 
-  describe('when there are known errors', () => {
-    let props;
-    let wrapper;
-    beforeEach(() => {
-      props = createTestProps({ touched: true, errors: { missing: true } });
-      wrapper = shallow(<TextField {...props} />);
-    });
-    it('should mark the TextInput as erroneous', () => {
-      expect(wrapper.find(TextInput)).toHaveProp('hasError', true);
-    });
-    it('should render the known error', () => {
-      expect(wrapper).toRender(FieldErrors);
+describe('when required', () => {
+  it('should add `*` to title`', () => {
+    const { getByText } = renderTextField({ isRequired: true });
+    expect(getByText('*')).toBeInTheDocument();
+  });
+});
+
+describe('when showing an info button', () => {
+  it('should render an info button', () => {
+    const onInfoButtonClick = jest.fn();
+    const { getByLabelText } = renderTextField({
+      onInfoButtonClick,
     });
+    expect(getByLabelText('More Info')).toBeInTheDocument();
+  });
+  it('should call onInfoButtonClick when button is clicked', () => {
+    const onInfoButtonClick = jest.fn();
+    const { container } = renderTextField({ onInfoButtonClick });
+    container.querySelector('button').click();
+    expect(onInfoButtonClick).toHaveBeenCalled();
   });
+});
 
-  describe('when there are unknown errors', () => {
-    let props;
-    let wrapper;
-    beforeEach(() => {
-      props = createTestProps({
+describe('when field is touched and has errors', () => {
+  describe('when field empty', () => {
+    it('should render a default error', () => {
+      const { getByText } = renderTextField({
         touched: true,
-        renderError: jest.fn(key => key),
-        errors: { customError: 5 },
+        errors: { missing: true },
       });
-      wrapper = shallow(<TextField {...props} />);
-    });
-    it('should mark the NumberInput as erroneous', () => {
-      expect(wrapper.find(TextInput)).toHaveProp('hasError', true);
+      expect(
+        getByText('This field is required. Provide a value.')
```

Comment:
A tip to keep this test working even when the message changes would be to use a RegEx: `getByText(/field is required/i)`.